### ISSUE TITLE
Generating deployment script, fails due to invalid path

### ISSUE THREAD

**Issue by wmmihaa** [Azure/webapps-deploy#196]
## Background
I raised this issue before (#42), but as it took too long to resolve, I moved on to other solutions. As I'm now working on a new project, I though I give it another chance, sadly to discover the issue has yet not been resolved.

## Solution set up:
I have a solution with ASP.Net web app together with a few other projects. All projects are .Net5 except one which is .Net3.1 (Azure Functions). All project folders resides under the same solution folder:
- Solution folder
   - Project folder
   - Project folder
   - Project folder

## Problem 
I'm trying to deploy the web application to an Azure Web App Slot (dev). So far I have only tried it using a Windows host.

I've tried generate my deployment.yml file both from Azure portal and from with in Visual Studio 2019, both with the same issue related to the `azure/webapps-deploy@v2` step.

```bash
Package deployment using ZIP Deploy initiated.
Updating submodules.
Preparing deployment for commit id '09ee5b10f6'.
Generating deployment script.
Using the following command to generate deployment script: 'azure site deploymentscript -y --no-dot-deployment -r "C:\local\Temp\zipdeploy\extracted" -o "C:\home\site\deployments\tools" --basic --sitePath "C:\local\Temp\zipdeploy\extracted\GreenEdge.Web"'
```

**The problem is the `sitePath`. The `extracted` directory holds all the binaries (build output) and there is no GreenEdge.Web directory (btw "GreenEdge.Web" is the name of  the project and the folder it's in).**. (the project output directory is "D:\a\GreenEdge\GreenEdge\GreenEdge.Web\publish")

## Logs and build scripts
I've attached the logs, and the yml is found below:
[build.log](https://github.com/Azure/webapps-deploy/files/7181117/build.log)

```yaml
name: Build and deploy .NET Core app to Windows WebApp greenedge-web
on:
  push:
    branches:
    - dev
env:
  AZURE_WEBAPP_NAME: greenedge-web
  AZURE_WEBAPP_SLOT_NAME: dev
  AZURE_WEBAPP_PACKAGE_PATH: GreenEdge.Web/publish
  AZURE_WEBAPP_PUBLISH_PROFILE: ${{ secrets.greenedge_web_dev_75dd }}
  CONFIGURATION: Release
  DOTNET_CORE_VERSION: 5.0.x
  WORKING_DIRECTORY: GreenEdge.Web
jobs:
  build-and-deploy:
    runs-on: windows-latest
    steps:
    - uses: actions/checkout@v2
    - name: Setup .NET Core
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: ${{ env.DOTNET_CORE_VERSION }}
    - name: Restore
      run: dotnet restore "${{ env.WORKING_DIRECTORY }}"
    - name: Build
      run: dotnet build "${{ env.WORKING_DIRECTORY }}" --configuration ${{ env.CONFIGURATION }} --no-restore
    - name: Test
      run: dotnet test "${{ env.WORKING_DIRECTORY }}" --no-build
    - name: Publish
      run: dotnet publish "${{ env.WORKING_DIRECTORY }}" --configuration ${{ env.CONFIGURATION }} --no-build --output "${{ env.AZURE_WEBAPP_PACKAGE_PATH }}"
    - name: List package path
      run: ls "${{ env.AZURE_WEBAPP_PACKAGE_PATH }}"
    - name: Deploy to Azure WebApp
      uses: azure/webapps-deploy@v2
      with:
        app-name: ${{ env.AZURE_WEBAPP_NAME }}
        package: ${{ env.AZURE_WEBAPP_PACKAGE_PATH }}
        publish-profile: ${{ env.AZURE_WEBAPP_PUBLISH_PROFILE }}
        slot-name: ${{ env.AZURE_WEBAPP_SLOT_NAME }}
    - name: Publish Artifacts
      uses: actions/upload-artifact@v1.0.0
      with:
        name: webapp
        path: ${{ env.AZURE_WEBAPP_PACKAGE_PATH }}

```

**Comment by wmmihaa** [comment]
## UPDATE
After setting up a new site I created a new deployment from VS, and it worked. I also created a slot and could deploy to that one too.
The difference with this deployment was that I started with creating the deployment from VS, while I created the deployment from the Azure portal the first time.

I will leave the issue open, as I still think this is an issue which needs to be resolved.

**Comment by BALAGA-GAYATRI** [comment]
Hey @wmmihaa, This issue will be investigated and you will be updated with appropriate solution.

**Comment by spellegrino021** [comment]
Hi @wmmihaa, when you say you started a deployment from the azure portal did you go to the Deployment Center in the Azure portal and configure your deployment from there? If so, do you mind providing the logs from the portal as well? You could also try modifying your workflow file to use path: ${{env.DOTNET_ROOT}}/myapp in the publish step, it looks like from the logs here it cannot find the build. Attached portal example workflow file for context on path update.

****_Error: Could not find a part of the path 'C:\local\Temp\zipdeploy\extracted\GreenEdge.Web'.
2021-09-16T20:13:51.5473123Z KuduSync.NET from: 'C:\local\Temp\zipdeploy\extracted\GreenEdge.Web' to: 'C:\home\site\wwwroot'
2021-09-16T20:13:51.5475428Z Failed exitCode=1, command="kudusync" -v 50  -f "C:\local\Temp\zipdeploy\extracted\GreenEdge.Web" -t "C:\home\site\wwwroot" -n "C:\home\site\deployments\09ee5b10f6e547b99a8ac90e64190486\manifest" -p "C:\home\site\deployments\a08bfb0853484c3d8e57fcc762a66269\manifest" -i ".git;.hg;.deployment;deploy.cmd"
2021-09-16T20:13:51.5477305Z An error has occurred during web site deployment._****

on:
  push:
    branches:
      - master
  workflow_dispatch:

jobs:
  build:
    runs-on: windows-latest

    steps:
      - uses: actions/checkout@v2

      - name: Set up .NET Core
        uses: actions/setup-dotnet@v1
        with:
          dotnet-version: '5.0.x'
          include-prerelease: true

      - name: Build with dotnet
        run: dotnet build --configuration Release

      - name: dotnet publish
        run: dotnet publish -c Release -o ${{env.DOTNET_ROOT}}/myapp

      - name: Upload artifact for deployment job
        uses: actions/upload-artifact@v2
        with:
          name: .net-app
          path: ${{env.DOTNET_ROOT}}/myapp

  deploy:
    runs-on: windows-latest
    needs: build
    environment:
      name: 'production'
      url: ${{ steps.deploy-to-webapp.outputs.webapp-url }}

    steps:
      - name: Download artifact from build job
        uses: actions/download-artifact@v2
        with:
          name: .net-app

      - name: Deploy to Azure Web App
        id: deploy-to-webapp
        uses: azure/webapps-deploy@v2
        with:
          app-name: 'appname'
          slot-name: 'dev'
          publish-profile: ${{ secrets.AzureAppService_PublishProfile_1234 }}
          package: .

**Comment by wmmihaa** [comment]
>Hi @wmmihaa, when you say you started a deployment from the azure portal did you go to the Deployment Center in the Azure portal and configure your deployment from there?

Yes I did, however I don't have any logs at this point. This is a production environment and as I said before, it's now working so I don't want to mess with it right now (I'm sure you understand).

**Comment by wmmihaa** [comment]
### If anyone is still getting this error, here is the solution and a desctiption of what likely caused the problem.

Before GitHub actions were a thing, we used Kudo directly from the Deployment Center of the App Service. However, becase our solution had multiple projects, we had to set a `PROJECT` configuration variable so Kudo would understand which project to deploy.

Since then, we have copied our app service configuration every time we set up a new site, including the `PROJECT` configuration variable. This variable is not required when using GitHub Actions, but is still used in Kudo, which is still the underlying framework for deploying your app when using `azure/webapps-deploy@v2`.

With the  `PROJECT` configuration variable set, Kudo will add it as a parameter as visible in the log:
```bash
Using the following command to generate deployment script: 'azure site deploymentscript -y --no-dot-deployment -r "C:\local\Temp\zipdeploy\extracted" -o "C:\home\site\deployments\tools" --basic --sitePath "C:\local\Temp\zipdeploy\extracted\YOUR-PROJECT-VARIABLE"'.
```

In conclusion, just remove the  `PROJECT` configuration variable from your App Service Configuration.
